### DISCUSSION TITLE
Normalize pixels to [0,1] using im2rec or reassign values to rec files

### DISCUSSION THREAD

**Discussion by lichen11** [apache/incubator-mxnet#13139] [Q&A]
I want to normalize the pixel values into [0, 255] without reading the images, normalizing them and re-saving them as jpeg. I used the im2rec API in R and notice there is no value normalization in this API. Then I tried reassign values to the rec files as follows:

    val = mx.io.ImageRecordIter(
            path.imgrec = val_data,
            batch.size  = batch_size,
            data.shape  = data_shape,
            rand.crop   = FALSE,
            rand.mirror = FALSE)
    val$reset()
    val$iter.next()
    val_data = val$value()$data
    val_data = as.array(val_data)
    val_data = val_data/255
    val$value()$data = val_data

I got the error below
    Error in val$value()$data = val_data : 
      invalid (NULL) left side of assignment

Is there another way to normalize the images by 255 in mxnet R? Thank you!

**Comment by leleamol** [comment]
@lichen11 
For questions, please submit on MXNet discussion forum (https://discuss.mxnet.io), where it will get a wider audience and allow other to learn as well.
 I'm closing this issue now in favor of the discussion forum issue you will file, please feel free to re-open if closed in error.
 Thanks!

@mxnet-label-bot [R, Question]

**Comment by lichen11** [comment]
Hi, I actually also noticed an error in the MXNet R API im2rec. When the original images are actually normalized to [0,1], when using im2rec to pack the images, the pixel values are scaled to [0, 255].
